File: frontend/lib/types/topics.ts
export interface TopicOut {
  id: string;
  institution_id: string | null;
  parent_id: string | null;
  name: string;
  slug: string;
  path: string;
  level: number;
  description: string | null;
  is_active: boolean;
  display_order: number;
  created_by_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface TopicTree extends TopicOut {
  children: TopicTree[];
}

export interface TopicCreate {
  name: string;
  slug: string;
  parent_id?: string | null;
  description?: string | null;
  display_order?: number;
}

export interface TopicUpdate {
  name?: string;
  slug?: string;
  description?: string | null;
  display_order?: number;
  is_active?: boolean;
}

File: ios/Synapse/Models/Topics.swift
import Foundation

struct TopicOut: Codable, Identifiable, Sendable, Hashable {
    let id: String
    let institutionId: String?
    let parentId: String?
    let name: String
    let slug: String
    let path: String
    let level: Int
    let description: String?
    let isActive: Bool
    let displayOrder: Int
    let createdById: String?
    let createdAt: Date
    let updatedAt: Date
}

/// Recursive tree node (`GET /topics/tree`).
struct TopicTree: Codable, Identifiable, Sendable, Hashable {
    let id: String
    let institutionId: String?
    let parentId: String?
    let name: String
    let slug: String
    let path: String
    let level: Int
    let description: String?
    let isActive: Bool
    let displayOrder: Int
    let createdById: String?
    let createdAt: Date
    let updatedAt: Date
    let children: [TopicTree]
}

struct TopicCreate: Encodable, Sendable {
    var name: String
    var slug: String
    var parentId: String?
    var description: String?
    var displayOrder: Int?
}

struct TopicUpdate: Encodable, Sendable {
    var name: String?
    var slug: String?
    var description: String?
    var displayOrder: Int?
    var isActive: Bool?
}
